src/main/java/de/tum/cit/aet/artemis/programming/service/ParticipationVcsAccessTokenService.java

```diff
@@ -66,7 +73,12 @@ public ParticipationVCSAccessToken findByUserIdAndParticipationIdOrElseThrow(lon
     public ParticipationVCSAccessToken createVcsAccessTokenForUserAndParticipationIdOrElseThrow(User user, long participationId) {
         participationVcsAccessTokenRepository.findByUserIdAndParticipationIdAndThrowIfExists(user.getId(), participationId);
         var participation = programmingExerciseStudentParticipationRepository.findByIdElseThrow(participationId);
-        return createParticipationVCSAccessToken(user, participation);
+        if (participation.isOwnedBy(user)) {
+            return createParticipationVCSAccessToken(user, participation);
+        }
+        else {
+            throw new AccessForbiddenException("Participation not owned by user");
```

Comment:
Do we need to add a translation here?

Reply:
No, as this error is never visible to the end user. The client just sets a variable if the user does not own the participation. This avoids that the instructors get tokens for every student's participation

See code-button.component.ts line 237